File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_map<int, int> freq;
        int n = arr.size();

        for (int i = 0; i < n; ++i) {
            int current = arr[i];
            if (freq.count(current * 2) || (current % 2 == 0 && freq.count(current / 2))) {
                return true;
            } else {
                freq[current]++;
            }
        }

        return false;
    }
};

File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        n = len(arr)
        d = defaultdict(int)

        for i in range(n):
            if arr[i]*2 in d or arr[i]/2 in d:
                return True
            else :
                d[arr[i]] += 1 

        return False
